spack/packages/2decomp-fft/package.py

```diff
@@ -20,15 +20,46 @@ class _2decompFft(MakefilePackage):
 
     version('1.5.847', sha256='b137d7cf9b771de0a174d1e6c4ff0e48b4a84b51ff6c62140b4f522092e6784f')
 
+    variant(
+        'backend', default='generic', description='FFT backend',
+        values=('generic', 'fftw3', 'mkl'), multi=False
+    )
+
     depends_on('mpi')
+    depends_on('mkl', when='backend=mkl')
+    depends_on('fftw', when='backend=fftw3')
 
-    @property
-    def build_targets(self):
-        return ['-j1']
+    #@property
+    #def build_targets(self):
+    #    return ['-j1']
 
     def edit(self, spec, prefix):
-        with working_dir('src'):
-            copy('Makefile.inc.x86', 'Makefile.inc')
+        makefile = FileFilter('src/Makefile')
+        makefile.filter('include Makefile.inc', '')
+        #with working_dir('src'):
+        #    touch('Makefile.inc')
+
+    def build(self, spec, prefix):
+        make_args = ['-j1', # This avoids a race condition in the Makefile
+                     'FFT={0}'.format(spec.variants['backend'].value),
+                     'LDFLAGS=-O3',
+                     'CC=mpicc',
+                     'F90=mpif90',
+                     'IFORT=gfortran',
+                     'OPTIONS=-DDOUBLE_PREC'
+                    ]
+        f90flags = 'F90FLAGS=-O3 -fcray-pointer -cpp'
```

Comment:
I thought they stopped using Cray string struct in 1990s.

Reply:
My objective was just mainly reproduce what they had recommended in their Makefile.inc.x86, so I haven't tested whether all options are necessary.  It does seem to build when I remove cray-pointer, so I guess it is safe to leave that out? Is there a scenario where someone is using an older fortran compiler where this would be needed?

Reply:
Looking at the source code, `SHM` macro enables use of Cray Fortran extension that is honored by GCC: https://gcc.gnu.org/onlinedocs/gfortran/Cray-pointers.html. I'd let it slide and allow unless someone enables Sytem V shared memory between processes.